tap/tcp_reader.go

```diff
@@ -81,14 +79,23 @@ func (reader *tcpReader) sendMsgIfNotClosed(msg api.TcpReaderDataMsg) {
 	reader.Unlock()
 }
 
+func (reader *tcpReader) isProtocolIdentified() bool {
+	return reader.parent.protocol != nil
+}
+
 func (reader *tcpReader) Read(p []byte) (int, error) {
+	if reader.exhaustBuffer {
+		reader.data = reader.buffer
+		reader.exhaustBuffer = false
+	}
+
 	var msg api.TcpReaderDataMsg
 
 	ok := true
 	for ok && len(reader.data) == 0 {
 		msg, ok = <-reader.msgQueue
 		if msg != nil {
-			reader.data = msg.GetBytes()
+			reader.data = append(reader.data, msg.GetBytes()...)
```

Comment:
Why append to `reader.data`? Its length is `0` here.

Reply:
@nimrod-up9 fixed with https://github.com/up9inc/mizu/commit/6eb29988b7155f75053c38bbff57344745773566